ext/spl/spl_iterators.c

```diff
@@ -2258,28 +2284,40 @@ static inline void spl_caching_it_next(spl_dual_it_object *intern)
 					return;
 				}
 			} else {
-				if (zend_is_true(&retval)) {
-					zend_call_method_with_0_params(Z_OBJ(intern->inner.zobject), intern->inner.ce, NULL, "getchildren", &zchildren);
+				bool has_children = zend_is_true(&retval);
+				zval_ptr_dtor(&retval);
+
+				if (has_children) {
+					zval args[2];
+
+					/* Store the children in the first constructor argument */
+					zend_call_method_with_0_params(Z_OBJ(intern->inner.zobject), intern->inner.ce, NULL, "getchildren", &args[0]);
 					if (EG(exception)) {
-						zval_ptr_dtor(&zchildren);
+						zval_ptr_dtor(&args[0]);
 						if (intern->u.caching.flags & CIT_CATCH_GET_CHILD) {
 							zend_clear_exception();
 						} else {
-							zval_ptr_dtor(&retval);
 							return;
 						}
 					} else {
-						ZVAL_LONG(&zflags, intern->u.caching.flags & CIT_PUBLIC);
-						spl_instantiate_arg_ex2(spl_ce_RecursiveCachingIterator, &intern->u.caching.zchildren, &zchildren, &zflags);
-						zval_ptr_dtor(&zchildren);
-					}
-				}
-				zval_ptr_dtor(&retval);
-				if (EG(exception)) {
-					if (intern->u.caching.flags & CIT_CATCH_GET_CHILD) {
-						zend_clear_exception();
-					} else {
-						return;
+						ZVAL_LONG(&args[1], intern->u.caching.flags & CIT_PUBLIC);
+
+						/* Instantiate object and call constructor */
```

Comment:
Pointless comment?
Preferably explain why you're doing something, not what (unless it's big blocks of code). In this case the function name and the comment say the same thing in different wordings.